arch/arc/Kconfig

```diff
@@ -278,13 +286,18 @@ config CODE_DENSITY
 	  Enable code density option to get better code density
 
 config ARC_HAS_ACCL_REGS
-	bool "Reg Pair ACCL:ACCH (FPU and/or MPY > 6)"
-	default y if CPU_HS3X || CPU_HS5X || CPU_HS6X
+	bool "Reg Pair ACCL:ACCH (FPU and/or MPY > 6 and/or DSP)"
+	default y if CPU_HS3X || CPU_HS4X || CPU_HS5X || CPU_HS6X
 	help
 	  Depending on the configuration, CPU can contain accumulator reg-pair
 	  (also referred to as r58:r59). These can also be used by gcc as GPR so
 	  kernel needs to save/restore per process
 
+config ARC_HAS_DSP
```

Comment:
We already have ARC_HAS_DSP Kconfig symbol:

https://github.com/zephyrproject-rtos/zephyr/blob/main/arch/arc/core/dsp/Kconfig#L6